preprocessor/src/main.rs

```diff
@@ -0,0 +1,279 @@
+use diesel::PgConnection;
+use dotenvy::dotenv;
+use feature_database::{imagedb, keypointdb, imagedb::ImageDatabase, keypointdb::KeypointDatabase, models};
+use feature_extraction::{akaze_keypoint_descriptor_extraction_def, DbKeypoints};
+use geotiff_lib::image_extractor;
+use geotiff_lib::image_extractor::{Datasets, MosaicDataset, MosaicedDataset};
+use homographier::homographier::raster_to_mat;
+use indicatif::{MultiProgress, ProgressBar};
+use tempfile::tempdir;
+
+use level_of_detail::calculate_amount_of_levels;
+use rayon as raycon;
+use raycon::Scope;
+
+use clap::Parser;
+use std::sync::{Arc, Mutex};
+
+pub mod level_of_detail;
+
+#[derive(Parser, Debug)]
+#[command(version, about, long_about = None)]
+struct Args {
+    /// The path to the folder where datasets are stored on disk
+    #[arg(short, long)]
+    dataset_path: Option<String>,
+
+    /// The path to a precomputed mosaic
+    #[arg(short, long)]
+    mosaic_path: Option<String>,
+
+    /// The path to the temp folder which shall contain processed files (/tmp will be used if not provided)
+    #[arg(long)]
+    temp_path: Option<String>,
+
+    /// The database url to connect to. Can also be provided by setting environment variable: DATABASE_URL
+    #[arg(long)]
+    database_url: Option<String>,
+
+    /// The tile size which the reference image will be split into
+    #[arg(short, long, default_value_t = 1000)]
+    tile_size: u64,
+
+    /// The number of CPU threads to use for processingcarg
+    #[arg(short, long, default_value_t = 1)]
+    cpu_num: usize,
+}
+
+type DbType = Arc<Mutex<PgConnection>>;
+
+fn main() {
+    dotenv().expect("Could not read .env file");
+
+    let args = Args::parse();
+
+    // Exit early if no path is provided.
+    if args.dataset_path.is_none() && args.mosaic_path.is_none() {
+        println!("No path provided to a dataset.");
+        std::process::exit(1);
+    }
+
+    // Must be in mutex since diesel is a sync library.
+    let db_connection: Arc<Mutex<PgConnection>> =
```

Comment:
```suggestion
    let db_connection: DbType =
```
når nu du har oprettet typen :)